### PR TITLE
Rustdoc UI fixes

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
The first commit fixes this bug (I couldn't figure out why we were setting the width manually and it works as expected without so...):

Screenshot from 2021-01-31 12-58-46

The second commit fixes a small bug. On tablets or computer with very little width, the search section goes "over" the search input, making it impossible to click on the search input:

Screenshot from 2021-01-31 13-22-37

The third and last commit fixes two bugs that you can see in this screenshot:

Screenshot from 2021-01-31 13-41-05

The wheel is going over the search input and the search tab is going under the search results text. The bug was fixed by simply switching to "mobile mode" at a bigger width:

Screenshot from 2021-01-31 13-49-50

cc @pickfire 
r? @Nemo157

### CODE REVIEW

File: src/librustdoc/html/static/rustdoc.css [hunk 1]
```diff
@@ -1578,7 +1578,7 @@ h4 > .notable-traits {
 	}
 }
 
-@media (max-width: 416px) {
+@media (max-width: 464px) {
```
Comment (pickfire) [inline src/librustdoc/html/static/rustdoc.css:1581]: Feels like a hacky solution to me, what if the width is slightly smaller or larger? Ideally we would want something to do with dynamic width like `calc` but that would also make stuff more complicated.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/rustdoc.css:1581]: It actually depends on the min width of the search input mostly. ;)

**Comment by pickfire** [comment]
image

The nightly title text looks off.

**Review by pickfire — APPROVED** [review]
Looks good to me other than the nightly text issue (that could be fixed in different commit).

File: src/librustdoc/html/static/rustdoc.css (hunk 1 above)
Comment (pickfire) [reply src/librustdoc/html/static/rustdoc.css:1581]: Other than that, on mobile, once you click on the menu, the body jumps off, that behavior looks weird.

**Comment by GuillaumeGomez** [comment]
@pickfire: I think this PR is already "diverse" enough. Can you open an issue for the text problem please?

File: src/librustdoc/html/static/rustdoc.css (hunk 1 above)
Comment (Nemo157) [reply src/librustdoc/html/static/rustdoc.css:1581]: Do you mean the body jumping down when you click the hamburger? That appears to be a nightly regression, it doesn't happen on the stable docs.

**Pull request merged**